### PR TITLE
Refactor finish_lesson to find ActivitySessions one time only

### PR DESCRIPTION
**Pull request by cissyxyu**
## WHAT
We're having performance issues on this endpoint because we're making the same call `ActivitySession.find()` four different times on different methods.

Instead of calling these methods so many times, I'm refactoring the code to call this one time and then pass the resulting array through to the helper methods.

## WHY
This will hopefully speed up this endpoint.

## HOW
Only make one call to the database to find all the ActivitySessions. Then pass these sessions through to the helper methods instead of making each helper method replicate the same call again.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Performance-Api-V1-ClassroomUnitsController-finish_lesson-e6181d777ce64742b484584bb23cef84

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | Not yet - deploying now!
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | Yes

### CODE REVIEW

File: services/QuillLMS/app/models/activity_session.rb [hunk 1]
```diff
@@ -389,13 +383,8 @@ def self.save_concept_results(classroom_unit_id, activity_id, concept_results)
...
 
-  def self.delete_activity_sessions_with_no_concept_results(classroom_unit_id, activity_id)
+  def self.delete_activity_sessions_with_no_concept_results(activity_sessions)
     incomplete_activity_session_ids = []
-    activity = Activity.find_by_id_or_uid(activity_id)
-    activity_sessions = ActivitySession.where(
-      classroom_unit_id: classroom_unit_id,
-      activity: activity
-    )
     activity_sessions.each do |as|
```
Comment (happythenewsad) [inline services/QuillLMS/app/models/activity_session.rb:388]: Optional nit: you could make the previous code a little tighter with `select` + `map`

File: services/QuillLMS/spec/controllers/api/v1/classroom_units_controller_spec.rb [hunk 2]
```diff
@@ -75,6 +75,9 @@
   context '#finish_lesson' do
     let(:follow_up_activity) { create(:activity) }
     let(:activity) { create(:activity, follow_up_activity: follow_up_activity) }
+    # let(:student) { create(:student)}
```
Comment (happythenewsad) [inline services/QuillLMS/spec/controllers/api/v1/classroom_units_controller_spec.rb]: forgot to delete?

**Review by happythenewsad — APPROVED** [review]
Love it! But you should delete the commented out lines unless there's a strong reason to keep them.

**Review by emilia-friedberg — APPROVED** [review]

**Review by anathomical — APPROVED** [review]
Nice solution.

**Pull request merged**